search_username/catalog/views.py

```diff
@@ -0,0 +1,46 @@
+from django.shortcuts import render, redirect
+from InstagramAPI import InstagramAPI
+from search_username.settings import password, user_name
+
+
+class GetInstagramInfo:
+    def __init__(self, username, password):
+        self.api = InstagramAPI(username=username, password=password)
+        self.api.login()
+
+    def get_user_id(self, name):
+        """Returns id user."""
+        self.api.searchUsername(name)
+        return self.api.LastJson["user"]["pk"]
+
+    def get_followers(self, user_id=None):
+        """Returns list of user's followers."""
+        user_id = user_id or self.api.username_id
+        followers = self.api.getTotalFollowers(user_id)
+        return [follower['username'] for follower in followers]
+
+    def get_followings(self, user_id=None):
+        """Returns list of user's followings."""
+        user_id = user_id or self.api.username_id
+        followings = self.api.getTotalFollowings(user_id)
+        return [following['username'] for following in followings]
+
+
+instagram_info = GetInstagramInfo(user_name, password)
+
+
+def index(request):
+    user = None
+    if request.method == "GET":
+        name = request.GET.get('name', '')
```

Comment:
Do you really need a second parameter here?

Reply:
yes,
if there is no name, there will be an error

Reply:
Why?

Reply:
because if there is no name it will search for a user who does not exist

Reply:
and it will be a mistake

Reply:
Where it'll search for a non-existing user?

Reply:
in instagram

Reply:
Where? Tell me line of code

Reply:
works without a second parameter